src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,129 @@
+import java.util.Optional;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+import mate.academy.model.DataProcessingException;
+import mate.academy.util.ConnectionUtil;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+
+    @Override
+    public Book create(Book book) {
+        Connection connection = ConnectionUtil.getConnection();
+        String sqlInsert = "INSERT INTO books(title, price) VALUES (?, ?)";
+        try {
+            PreparedStatement statement = connection.prepareStatement(
+                    sqlInsert,
+                    Statement.RETURN_GENERATED_KEYS
+            );
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            int affectedRows = statement.executeUpdate();
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                book.setId(id);
+            }
+            if (affectedRows < 1) {
+                throw new RuntimeException("Book wasn't added. 0 rows were affected");
+            }
+            statement.close();
+            connection.close();
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Error while adding book", e);
+        }
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        Connection connection = ConnectionUtil.getConnection();
+        String sqlSelect = "SELECT * FROM books WHERE id = ?";
+        try {
+            PreparedStatement statement = connection.prepareStatement(sqlSelect);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (!resultSet.first()) {
+                return Optional.empty();
+            }
+            statement.close();
+            connection.close();
+            return Optional.of(getBookFromResultSet(resultSet));
+        } catch (SQLException e) {
+            throw new DataProcessingException("Error occurred wile finding by id", e);
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        Connection connection = ConnectionUtil.getConnection();
+        String sqlSelect = "SELECT * FROM books";
+        try {
+            PreparedStatement statement = connection.prepareStatement(sqlSelect);
+            ResultSet resultSet = statement.executeQuery();
+            List<Book> res = new ArrayList<>();
+            while (resultSet.next()) {
+                res.add(getBookFromResultSet(resultSet));
+            }
+            statement.close();
+            connection.close();
+            return res;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Error occurred while getting all books", e);
+        }
+    }
+
+    @Override
+    public Book update(Book book) {
+        Connection connection = ConnectionUtil.getConnection();
```

Comment:
you should use `try-with-resources` for both `Connection` and `PreparedStatement`